sklearn/manifold/_locally_linear.py

```diff
@@ -18,17 +18,17 @@
 from ..neighbors import NearestNeighbors
 
 
-def barycenter_weights(X, Z, reg=1e-3):
-    """Compute barycenter weights of X from Y along the first axis
+def barycenter_weights(X, indices, reg=1e-3):
+    """Compute barycenter weights of X from indices along the first axis
 
-    We estimate the weights to assign to each point in Y[i] to recover
+    We estimate the weights to assign to each point in X[indices][i] to recover
     the point X[i]. The barycenter weights sum to 1.
 
     Parameters
     ----------
     X : array-like, shape (n_samples, n_dim)
```

Comment:
For consistency with the usual notation in scikit-learn (assuming n_dim is no longer used elsewhere in the docstring of this function):

```suggestion
    X : array-like, shape (n_samples, n_features)
```